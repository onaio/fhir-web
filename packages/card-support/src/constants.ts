// OPENSRP API endpoints
export const OPENSRP_URL_USER_ASSIGNMENT = '/organization/user-assignment';
export const OPENSRP_URL_CLIENT_SEARCH = '/client/search';
export const OPENSRP_URL_LOCATION_HIERARCHY = '/location/hierarchy';

/** CSV File MIME Types */
export const TEXT_CSV = 'text/csv';
export const APPLICATION_CSV = 'application/csv';
export const TEXT_PLAIN = 'text/plain';
export const TEXT_X_CSV = 'text/x-csv';
export const APPLICATION_VND_EXCEL = 'application/vnd.ms-excel';
export const APPLICATION_X_CSV = 'application/x-csv';
export const TEXT_COMMA_SEPARATED_VALUES = 'text/comma-separated-values';
export const TEXT_X_COMMA_SEPARATED_VALUES = 'text/x-comma-separated-values';
export const TEXT_TAB_SEPARATED_VALUES = 'text/tab-separated-values';
