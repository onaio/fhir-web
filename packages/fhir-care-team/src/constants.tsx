// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';
