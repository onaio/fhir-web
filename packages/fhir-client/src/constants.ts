export const dateFormat = 'yyyy-MM-dd';
