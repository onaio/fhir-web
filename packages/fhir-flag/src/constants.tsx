// urls
export const URL_ADMIN = '/admin';
export const URL_CLOSE_FLAGS = `${URL_ADMIN}/close-flags`;

// // magic strings

export const thatiMinutes = 30 * 60 * 1000;

// form fields
export const locationName = 'locationName' as const;
export const productName = 'productName' as const;
export const status = 'status' as const;
export const comments = 'comments' as const;
