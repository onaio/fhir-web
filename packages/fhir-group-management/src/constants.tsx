// url
export const ADD_EDIT_GROUP_URL = '/group/add-edit';
export const ADD_EDIT_COMMODITY_URL = '/commodity/add-edit';
export const LIST_GROUP_URL = '/groups/list';
export const LIST_COMMODITY_URL = '/commodity/list';

// magic strings
export const groupResourceType = 'Group';
export const listResourceType = 'List';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const type = 'type' as const;
export const unitOfMeasure = 'unitOfMeasure' as const;
