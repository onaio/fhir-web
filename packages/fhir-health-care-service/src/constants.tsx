// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTH_CARE_SERVICE_RESOURCE_TYPE = 'HealthcareService';
export const HEALTHCARES_ENDPOINT = 'HealthcareService/';
