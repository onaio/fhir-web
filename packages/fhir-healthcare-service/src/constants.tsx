// url
export const ADD_EDIT_HEALTHCARE_SERVICE_URL = '/healthcare/add-edit';
export const LIST_HEALTHCARE_URL = '/healthcare/list';

//
// constants
//
export const healthCareServiceResourceType = 'HealthcareService';

// form constants
export const id = 'id' as const;
export const identifier = 'identifier' as const;
export const name = 'name' as const;
export const active = 'active' as const;
export const comment = 'comment' as const;
export const extraDetails = 'extraDetails' as const;
export const providedBy = 'providedBy' as const;
