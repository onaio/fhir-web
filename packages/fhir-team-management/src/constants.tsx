// url
export const URL_ADMIN = 'admin';
export const URL_LIST_TEAMS = `${URL_ADMIN}/teams/`;
export const URL_ADD_TEAM = 'teams/add/';
export const URL_EDIT_TEAM = 'teams/edit/';

//
// Practitioner
//
export const PRACTITIONERROLE_GET = 'PractitionerRole/';
export const PRACTITIONERROLE_POST = 'PractitionerRole/';
export const PRACTITIONERROLE_DEL = 'PractitionerRole/';

//
// Practitioner Role
//
export const PRACTITIONER_GET = 'Practitioner/';
export const PRACTITIONER_POST = 'Practitioner/';
export const PRACTITIONER_DEL = 'Practitioner/';

//
// Teams
//
export const TEAMS_GET = 'Organization/';
export const TEAMS_PUT = 'Organization/';
export const TEAMS_POST = 'Organization/';
