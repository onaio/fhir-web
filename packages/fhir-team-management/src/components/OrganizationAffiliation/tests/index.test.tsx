import { AffiliationList } from '..';
import React from 'react';
import { Route, Router, Switch } from 'react-router';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { store } from '@opensrp/store';
import nock from 'nock';
import { cleanup, render, screen, waitFor, within } from '@testing-library/react';
import { fireEvent, waitForElementToBeRemoved } from '@testing-library/dom';
import { createMemoryHistory } from 'history';
import { authenticateUser } from '@onaio/session-reducer';
import { locationHierarchyResourceType } from '@opensrp/fhir-location-management';
import {
  allAffiliations,
  allOrgs,
  createdAffiliation1,
  createdAffiliation2,
  fhirHierarchy,
} from './fixures';
import { organizationAffiliationResourceType, organizationResourceType } from '../../../constants';
import userEvent from '@testing-library/user-event';
import * as notifications from '@opensrp/notifications';
import { RoleContext } from '@opensrp/rbac';
import { superUserRole } from '@opensrp/react-utils';

jest.mock('@opensrp/notifications', () => ({
  __esModule: true,
  ...Object.assign({}, jest.requireActual('@opensrp/notifications')),
}));

jest.mock('fhirclient', () => {
  return jest.requireActual('fhirclient/lib/entry/browser');
});

const mockV4 = '75482e89-40ce-4512-a74a-ccc30aeae073';

jest.mock('uuid', () => {
  const v4 = () => mockV4;
  return {
    _esModule: true,
    ...Object.assign({}, jest.requireActual('uuid')),
    v4,
  };
});

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});

const props = {
  fhirBaseURL: 'http://test.server.org',
  fhirRootLocationId: 'rootLoc',
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AppWrapper = ({ children }: any) => {
  return (
    <Provider store={store}>
      <RoleContext.Provider value={superUserRole}>
        <QueryClientProvider client={queryClient}>
          <Switch>
            <Route exact path="/assignments">
              {children}
            </Route>
          </Switch>
        </QueryClientProvider>
      </RoleContext.Provider>
    </Provider>
  );
};

afterEach(() => {
  cleanup();
  nock.cleanAll();
});

beforeAll(() => {
  nock.disableNetConnect();
  store.dispatch(
    authenticateUser(
      true,
      {
        email: 'bob@example.com',
        name: 'Bobbie',
        username: 'RobertBaratheon',
      },
      { api_token: 'hunter2', oAuth2Data: { access_token: 'sometoken', state: 'abcde' } }
    )
  );
});

afterAll(() => {
  nock.enableNetConnect();
});

test('Edits organization affiliation correctly', async () => {
  // - Simulates conditions where we edit 2 organizationAffiliation like so:
  // - one organization affiliation is modified - starts with mapping an organization to several locations
  // - one organization affiliation is deleted - organization is no longer assigned to any location.

  const successMock = jest
    .spyOn(notifications, 'sendSuccessNotification')
    .mockImplementation(() => {
      return;
    });
  const history = createMemoryHistory();
  history.push('/assignments');

  nock(props.fhirBaseURL)
    .get(`/${locationHierarchyResourceType}/_search`)
    .query({ _id: props.fhirRootLocationId })
    .reply(200, fhirHierarchy);

  nock(props.fhirBaseURL)
    .get(`/${organizationAffiliationResourceType}/_search`)
    .query({ _summary: 'count' })
    .reply(200, { total: 1000 })
    .get(`/${organizationAffiliationResourceType}/_search`)
    .query({ _count: 1000 })
    .reply(200, allAffiliations);

  nock(props.fhirBaseURL)
    .get(`/${organizationResourceType}/_search`)
    .query({ _summary: 'count' })
    .reply(200, { total: 1000 })
    .get(`/${organizationResourceType}/_search`)
    .query({ _count: 1000 })
    .reply(200, allOrgs);

  const { unmount } = render(
    <Router history={history}>
      <AppWrapper>
        <AffiliationList {...props} />
      </AppWrapper>
    </Router>
  );

  await waitForElementToBeRemoved(document.querySelector('.ant-spin'));
  // now wait for table spinner
  await waitForElementToBeRemoved(document.querySelector('.ant-spin'));

  // initially show single user defined root location
  const treeSection = document.querySelector('.ant-tree') as HTMLElement;
  const firstRootLoc = within(treeSection).getByTitle(/Ona Office Sub Location/);
  expect(firstRootLoc).toMatchSnapshot('user defined root location');

  // table row should be one containing information
  const table = document.querySelector('table');
  // how many body rows - should be 1
  expect(table.querySelectorAll('tbody tr')).toHaveLength(1);
  const firstRowTd = table.querySelectorAll('tbody tr:nth-child(1) td');
  firstRowTd.forEach((td) => {
    expect(td).toMatchSnapshot('first row containing ona office loc details');
  });

  // open modal to start editing organizations
  // Edit btns could be many but I know for this page the table has only one row with an edit
  const editBtn = screen.getByText('Edit');
  expect(editBtn).toMatchInlineSnapshot(`
    <span>
      Edit
    </span>
  `);
  userEvent.click(editBtn);

  // look for Modal with select
  const selectInModal = screen.getByTestId('affiliation-select');
  selectInModal.querySelectorAll('.ant-select-selection-item-content').forEach((item) => {
    expect(item).toMatchSnapshot('default selected item');
  });

  // should have some options as default
  const input = selectInModal.querySelector('input');
  fireEvent.mouseDown(input);

  // see what org options are listed as available for selection
  const beforeFilterOptions = [...document.querySelectorAll('.ant-select-item-option')].map(
    (element) => {
      return element.textContent;
    }
  );
  expect(beforeFilterOptions).toEqual([
    'Test Team 3',
    'Test Team 4',
    'Test Team 5',
    'Test Team 5',
    'Voidsingers',
  ]);

  // try and filter
  await userEvent.type(input, 'void');
  // see what org options are listed as available for selection
  const afterFilterOptions = [...document.querySelectorAll('.ant-select-item-option')].map(
    (element) => {
      return element.textContent;
    }
  );
  expect(afterFilterOptions).toEqual(['Voidsingers']);

  // util to help remove selected options using the display value
  const removeSelectedOptionWithText = (textContent) => {
    //remove one of the selection test Team 3
    const testTeam3Span = screen.getByTitle((content, element) => {
      return content.includes(textContent) && element.tagName === 'SPAN';
    });
    const removeIcon = testTeam3Span.querySelector('.ant-select-selection-item-remove') as Element;
    fireEvent.click(removeIcon);
  };
  // remove all selected options for ona office sub location
  removeSelectedOptionWithText('Test Team 3');
  removeSelectedOptionWithText('Test Team 4');

  // select Voidsingers
  fireEvent.click(screen.getByTitle(/Voidsingers/));

  // only voidSingers is selected
  const selectedOptions = [...document.querySelectorAll('.ant-select-item-option-selected')].map(
    (option) => option.textContent
  );
  expect(selectedOptions).toEqual(['Voidsingers']);

  // then try and submit

  // nock queries at this point
  nock(props.fhirBaseURL)
    .put(`/${organizationAffiliationResourceType}/75482e89-40ce-4512-a74a-ccc30aeae073`, {
      ...createdAffiliation1,
      id: '75482e89-40ce-4512-a74a-ccc30aeae073',
    })
    .reply(201, {})
    .put(`/${organizationAffiliationResourceType}/${createdAffiliation2.id}`, createdAffiliation2)
    .reply(201, {})
    .delete(`/${organizationAffiliationResourceType}/1575`)
    .reply(200, {});

  // now submit
  const submitBtn = screen.getByTestId('submit-affiliations');
  fireEvent.click(submitBtn);

  await waitFor(() => {
    expect(successMock).toHaveBeenCalledWith('Team assignments updated successfully');
    expect(screen.queryByText(/Assign\/Unassign Teams/i)).not.toBeInTheDocument();
  });

  // lets expand on tree search and select a node
  const antTree = document.querySelector('.ant-tree-list');
  const firstDownCaret = antTree.querySelector('.anticon-caret-down');
  fireEvent.click(firstDownCaret);

  // tree title visible now - represents nested nodes that are now visible
  document.querySelectorAll('.ant-tree-title').forEach((nodeTitle) => {
    expect(nodeTitle).toMatchSnapshot('visible nodes on expand');
  });

  const lastTextNode = screen.getByText('Part Of Sub Location');
  fireEvent.click(lastTextNode);

  // tree title visible now - represents nested nodes that are now visible
  document.querySelectorAll('.ant-tree-title').forEach((nodeTitle) => {
    expect(nodeTitle).toMatchSnapshot('visible nodes on expanding part of sub location');
  });

  // table change
  expect(document.querySelectorAll('table tbody tr')).toHaveLength(5);

  expect(nock.pendingMocks()).toEqual([]);
  expect(nock.isDone()).toBeTruthy();

  await waitFor(() => {
    expect(successMock).toHaveBeenCalledWith('Team assignments updated successfully');
  });

  unmount();
});

test('api error response', async () => {
  const history = createMemoryHistory();
  history.push('/assignments');

  nock(props.fhirBaseURL)
    .get(`/${locationHierarchyResourceType}/_search`)
    .query({ _id: props.fhirRootLocationId })
    .replyWithError('Something awful happened');

  render(
    <Router history={history}>
      <AppWrapper>
        <AffiliationList {...props} />
      </AppWrapper>
    </Router>
  );

  await waitForElementToBeRemoved(document.querySelector('.ant-spin'));

  // error page
  expect(screen.getByText(/Something awful happened/)).toBeInTheDocument();
});

test('api undefined response', async () => {
  const history = createMemoryHistory();
  history.push('/assignments');

  const thisProps = {
    ...props,
    fhirRootLocationId: 'unknown',
  };

  nock(props.fhirBaseURL)
    .get(`/${locationHierarchyResourceType}/_search`)
    .query({ _id: thisProps.fhirRootLocationId })
    .reply(404, {
      resourceType: 'OperationOutcome',
      text: {
        status: 'generated',
      },
      issue: [
        {
          severity: 'error',
          code: 'processing',
          diagnostics: 'HAPI-2001: Resource Location/2252d is not known',
        },
      ],
    });

  render(
    <Router history={history}>
      <AppWrapper>
        <AffiliationList {...thisProps} />
      </AppWrapper>
    </Router>
  );

  await waitForElementToBeRemoved(document.querySelector('.ant-spin'));

  // error page
  expect(screen.getByText(/Resource Location\/2252d is not known/)).toBeInTheDocument();
});
