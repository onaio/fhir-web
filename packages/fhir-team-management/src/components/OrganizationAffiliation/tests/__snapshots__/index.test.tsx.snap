// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edits organization affiliation correctly: default selected item 1`] = `
<span
  class="ant-select-selection-item-content"
>
  Test Team 3
</span>
`;

exports[`Edits organization affiliation correctly: default selected item 2`] = `
<span
  class="ant-select-selection-item-content"
>
  Test Team 4
</span>
`;

exports[`Edits organization affiliation correctly: first row containing ona office loc details 1`] = `
<td
  class="ant-table-cell"
>
  Ona Office Sub Location
</td>
`;

exports[`Edits organization affiliation correctly: first row containing ona office loc details 2`] = `
<td
  class="ant-table-cell"
>
  Test Team 3, Test Team 4
</td>
`;

exports[`Edits organization affiliation correctly: first row containing ona office loc details 3`] = `
<td
  class="ant-table-cell"
>
  <button
    class="ant-btn css-dev-only-do-not-override-k7429z ant-btn-link action-button"
    type="button"
  >
    <span>
      Edit
    </span>
  </button>
</td>
`;

exports[`Edits organization affiliation correctly: user defined root location 1`] = `
<span
  title="Ona Office Sub Location"
>
  Ona Office Sub Location
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expand 1`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Ona Office Sub Location"
  >
    Ona Office Sub Location
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expand 2`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Level 0"
  >
    Level 0
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expand 3`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Arundel mobile clinic"
  >
    Arundel mobile clinic
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expand 4`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Part Of Sub Location"
  >
    Part Of Sub Location
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 1`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Ona Office Sub Location"
  >
    Ona Office Sub Location
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 2`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Level 0"
  >
    Level 0
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 3`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Arundel mobile clinic"
  >
    Arundel mobile clinic
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 4`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Part Of Sub Location"
  >
    Part Of Sub Location
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 5`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Nairobii"
  >
    Nairobii
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 6`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Test Sub Loc"
  >
    Test Sub Loc
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 7`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Test Loc 113"
  >
    Test Loc 113
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 8`] = `
<span
  class="ant-tree-title"
>
  <span
    title="Test Loc 1145"
  >
    Test Loc 1145
  </span>
</span>
`;

exports[`Edits organization affiliation correctly: visible nodes on expanding part of sub location 9`] = `
<span
  class="ant-tree-title"
>
  <span
    title="MOH HQ 2"
  >
    MOH HQ 2
  </span>
</span>
`;
